lib/Core/ExecutorUtil.cpp

```diff
@@ -126,6 +126,30 @@ namespace klee {
     if (numOperands > 1) op2 = evalConstant(ce->getOperand(1), ki);
     if (numOperands > 2) op3 = evalConstant(ce->getOperand(2), ki);
 
+    /* Checking for possible errors during constant folding */
+    switch (ce->getOpcode()) {
+    case Instruction::SDiv:
+    case Instruction::UDiv:
+    case Instruction::SRem:
+    case Instruction::URem:
+      if (op2->getLimitedValue() == 0) {
+	std::string msg("Division/modulo by zero during constant folding at location ");
+	llvm::raw_string_ostream os(msg);
+	os << (ki ? ki->printFileLine().c_str() : "[unknown]");
+	klee_error("%s", os.str().c_str());
```

Comment:
I'd say it's debatable. KLEE will evaluate all constants here, even if they are never used during execution. This would prevent someone running KLEE on a program that contains code like this even if it is completely irrelevant.

I think the best thing to do would be warn when evaluating constant expressions like this and generate the non-constant folded expression.

As mentioned previously though we can't do this right now due to the way constant folding is coupled with expression constructions. So this PR will do for now.